cpp/CMakeLists.txt

```diff
@@ -840,3 +841,79 @@ add_custom_command(OUTPUT CUDF_DOXYGEN
                    VERBATIM)
 
 add_custom_target(docs_cudf DEPENDS CUDF_DOXYGEN)
+
+
+####################################################################################################
+# - cuDatasource - OFF by default due to increased number of dependencies
+if(BUILD_CUDATASOURCE)
+
+    # cuDatasource library
+    add_library(libcudatasource
+        cudatasource/kafka/src/kafka_consumer.cpp
+    )
+
+    # Include paths
+    include_directories("${CMAKE_SOURCE_DIR}/cudatasource/kafka/include")
+    
+    # Rename installation to proper name for later finding
+    set_target_properties(libcudatasource PROPERTIES OUTPUT_NAME "cuDatasource")
+    set_target_properties(libcudatasource PROPERTIES BUILD_RPATH "\$ORIGIN")
+
+    ###################################################################################################
+    # cuDatasource - Kafka ------------------------------------------------------------------------------------
+
+    find_path(RDKAFKA_INCLUDE "librdkafka" HINTS "$ENV{RDKAFKA_ROOT}/include")
+    find_library(RDKAFKA++_LIBRARY "rdkafka++" HINTS "$ENV{RDKAFKA_ROOT}/lib" "$ENV{RDKAFKA_ROOT}/build")
+
+    message(STATUS "RDKAFKA: RDKAFKA++_LIBRARY set to ${RDKAFKA++_LIBRARY}")
+    message(STATUS "RDKAFKA: RDKAFKA_INCLUDE set to ${RDKAFKA_INCLUDE}")
+
+    target_link_libraries(libcudatasource ${RDKAFKA++_LIBRARY})
+
+    ###################################################################################################
+    # - cuDatasource Install ----------------------------------------------------------------------------------
+    target_link_libraries(cudf libcudatasource)
```

Comment:
libcudf shouldn't link against this, this should just link against libcudf

Reply:
Isn't that what this is doing? Linking `libcudf_kafka` against `cudf` so that `cudf` can invoke the custom datasource defined in `libcudf_kafka`?

Note: ^^ is after my latest pr with the updated names, github might show the original names so just wanted to point that out.